lib/web_of_science/data/records.rb

```diff
@@ -23,10 +24,25 @@ class Records
       # @return doc [Nokogiri::XML::Document] WOS records document
       attr_reader :doc
 
-      # @param xmlrecords [String] records in XML
+
+      # @param records [String] records in XML
       # @param encoded_records [String] records in HTML encoding
-      def initialize(xmlrecords: nil, encoded_records: nil)
-        @doc = WebOfScience::XmlParser.parse(xmlrecords, encoded_records)
+      def initialize(records: nil, encoded_records: nil)
+        @doc = parse(records, encoded_records)
+      end
+
+      # Return decoded XML, whether it is passed in already or needs to be decoded
+      # @param xml [String] XML
+      # @param encoded_xml [String] XML with HTML encoding
+      # @return [Nokogiri::XML::Document]
+      # @raise RuntimeError when arguments are all nil
+      def parse(xml, encoded_xml)
+        xml ||= begin
+          raise 'encoded_xml is nil' if encoded_xml.nil?
+          coder = HTMLEntities.new
+          coder.decode(encoded_xml)
+        end
+        Nokogiri::XML(xml) { |config| config.strict.noblanks }
       end
```

Comment:
This change will undo prior refactoring and break the code when it needs to work without an encoded record, if I understand this correctly.  I suggest this commit is dropped.